packages/ipns/src/dns-resolvers/resolver.ts

```diff
@@ -0,0 +1,25 @@
+import { Resolver } from 'dns/promises'
+import { CustomProgressEvent } from 'progress-events'
+import { resolveFn, type DNSResponse } from '../utils/dns.js'
+import type { DNSResolver } from '../index.js'
+
+const resolve: DNSResolver = async function resolve (domain, options = {}) {
+  const resolver = new Resolver()
+  const listener = (): void => {
+    resolver.cancel()
+  }
+
+  options.signal?.addEventListener('abort', listener)
+
+  try {
+    options.onProgress?.(new CustomProgressEvent<string>('dnslink:query', { detail: domain }))
+    const dnslinkRecord = await resolveFn(resolver, domain)
+
+    options.onProgress?.(new CustomProgressEvent<DNSResponse>('dnslink:answer', { detail: dnslinkRecord }))
```

Comment:
`dnslink` onProgress events were previously not being used in the nodejs resolve function. see https://github.com/ipfs/helia-ipns/blob/b87b9e3c3b8e54d3be6d9704c0598db5ba7d552c/packages/ipns/src/dns-resolvers/default.ts#L13-L39